load-testing/audit-service/Audit-service.md

```diff
@@ -1,53 +1,49 @@
 # How to run audit service load test
 
-### Step 1:
-The load test script mocks the POSTing of some logs on `presigned-url` and `login` endpoint, we need to turn off/ comment out
-the config (only for purpose of load testing) in `kube/services/revproxy/gen3.nginx.conf/audit-service.conf` in cloud-automation.
-This should be done from within the environment.
-```
-# only allow GET requests. POSTing audit logs is only for internal use
-    limit_except GET {
-      deny all;
-    }
-```
-The above configuration denies the user to POST logs on the endpoints in audit service
-
-### Step 2:
-`gen3 kube-setup-revproxy`
+### How is load Testing performed?
+User visits the commons and logs in to the via Incommons Login or Google ir requests a PresignedURL, a message is send to AWS SQS via Fence 
+which now sits between Fence and Audit Service. Audit service pulls in 10messages at a time from the queue to the audit DB. The load test will
+send a HTTP GET request to either `/audit/log/login` or `/audit/log/presigned_url` over a duration of 5 minutes with 100 virtual 
+users.
 
-### NOTE : After testing is done, turn on / uncomment the above configuration in audit-service.conf, and run STEP 2
+Before every run, the bash script will run `purge` command inside the bash script which will clear the SQS queue. So the
+load test will work with 0 messages in the queue.
 
-### Step 3:
+### Step 1:
 The user needs to be added to the user.yaml file with `audit_reader` policy
 
-### Step 4:
+### Step 2: if running the test locally
 Also, the user would need API key created in particular env which user needs to run audit service load testing in. 
-Download the `credential.json` file and paste in gen3-qa folder
+Download the `credential.json` file and paste in gen3-qa folder. 
+Run this ->
+```
+# for executing load test on presigned-url endpoint
+node load-testing/loadTestRunner.js credentials.json load-testing/sample-descriptors/load-test-audit-presigned-urls-sample.json
 
-### Step 5: Running the load tests
-You can either run via 
-- Jenkins job https://jenkins.planx-pla.net/job/gen3-run-load-tests/
+# for executing load test on login endpoint
+ node load-testing/loadTestRunner.js credentials.json load-testing/sample-descriptors/load-test-audit-login-sample.json
+```
 
-    How are above steps achieved in the Jenkins-job/config
+### Step 3: Running the load tests via Jenkins Job
+Jenkins job https://jenkins.planx-pla.net/job/gen3-run-load-tests/
+
+In the pre-test setup, we populate the AWS SQS with some fake messages that are send via bash script (location link TBA)
+
+```
+if [ "$LOAD_TEST_DESCRIPTOR" == "audit-presigned-url" ]; then
+  echo "Populating audit-service SQS with presigned-url messages"
+  chmod u+x gen3-qa/load-testing/audit-service/sendPresignedURLMessages.sh
```

Comment:
there is no need to chmod +x if this change is applied before committing the script to github.

Reply:
this change is made in Jenkins-job as well